devicetypes/smartthings/ecolink-water-freeze-sensor.src/ecolink-water-freeze-sensor.groovy

```diff
@@ -0,0 +1,191 @@
+/**
+ *  Copyright 2018 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ *  Ecolink Water/Freeze Sensor
+ *
+ */
+
+metadata {
+	definition(name: "Ecolink Water/Freeze Sensor", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "x.com.st.d.sensor.moisture", runLocally: false, minHubCoreVersion: '000.017.0012', executeCommandsLocally: false) {
+		capability "Water Sensor"
+		capability "Temperature Alarm"
+		capability "Sensor"
+		capability "Battery"
+		capability "Health Check"
+
+		fingerprint mfr: "014A", prod: "0005", model: "0010", deviceJoinName: "Ecolink Water/Freeze Sensor"
+	}
+
+	simulator {
+	}
+
+	tiles(scale: 2) {
+		multiAttributeTile(name: "water", type: "generic", width: 6, height: 4) {
+			tileAttribute("device.water", key: "PRIMARY_CONTROL") {
+				attributeState("dry", icon: "st.alarm.water.dry", backgroundColor: "#ffffff")
+				attributeState("wet", icon: "st.alarm.water.wet", backgroundColor: "#00A0DC")
+			}
+		}
+		valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "battery", label: '${currentValue}% battery', unit: ""
+		}
+		valueTile("temperatureAlarm", "device.temperatureAlarm", inactiveLabel: false, decoration: "flat", width: 4, height: 2) {
+			state "temperatureAlarm", label: 'Freeze: ${currentValue}', unit: ""
```

Comment:
based on this: https://github.com/SmartThingsCommunity/SmartThingsCapabilities/blob/master/capabilities/temperature-alarm.src/temperature-alarm.yml

shouldn't the states be "cleared" and "freeze"?

Reply:
Good idea.